### PR TITLE
[Fix](inverted index) remove IndexReader::indexExists, use fs interface

### PR DESCRIPTION
**Pull request by airborne12**
## Proposed changes

Issue Number: close #xxx

IndexReader::indexExists in CLucene will use system fs interface to find out whether a directory is exists, which is dangerous when writing files into S3, use fs->exists instead.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by airborne12** [comment]
run buildall

**Comment by zzzxl1993** [comment]
LGTM

File: be/src/olap/rowset/segment_v2/inverted_index_writer.cpp [hunk 1]
```diff
@@ -133,12 +133,18 @@ class InvertedIndexColumnWriterImpl : public InvertedIndexColumnWriter {
...
-                lucene::index::IndexReader::unlock(index_path.c_str());
+        bool exists = false;
+        auto st = _fs->exists(index_path.c_str(), &exists);
+        if (!st.ok()) {
+            LOG(ERROR) << "index_path:"
+                       << " exists error:" << st;
+            return st;
+        }
+        if (exists) {
+            st = _fs->delete_directory(index_path.c_str());
```
Comment (xiaokang) [inline be/src/olap/rowset/segment_v2/inverted_index_writer.cpp:144]: can just return error Status

**Comment by airborne12** [comment]
run buildall

**Review by xiaokang — APPROVED** [review]
LGTM

**Review by zzzxl1993 — APPROVED** [review]
LGTM

**Pull request merged**